src/clients/c++/examples/simple_cuda_shm_client.cc

```diff
@@ -0,0 +1,497 @@
+  err = input0->SetSharedMemory("input_data", 0, input_byte_size);
+  if (!err.IsOk()) {
+    std::cerr << "failed setting shared memory input: " << err << std::endl;
+    exit(1);
+  }
+  err = input1->SetSharedMemory("input_data", input_byte_size, input_byte_size);
+  if (!err.IsOk()) {
+    std::cerr << "failed setting shared memory input: " << err << std::endl;
+    exit(1);
+  }
+
+  // Send inference request to the inference server.
+  std::map<std::string, std::unique_ptr<nic::InferContext::Result>> results;
+  FAIL_IF_ERR(infer_ctx->Run(&results), "unable to run model");
+
+  int output0_data[16];
+  int output1_data[16];
+  ReadCUDAIPCHandle(output0_cuda_rep, output0_data)
+  ReadCUDAIPCHandle(output1_cuda_rep, output1_data)
+
+  // Client should use ipcCUDA_t handle to return data
+  // We expect there to be 2 results. Walk over all 16 result elements
+  // and print the sum and difference calculated by the model.
+  if (results.size() != 2) {
+    std::cerr << "error: expected 2 results, got " << results.size()
+              << std::endl;
+  }
+
+  for (size_t i = 0; i < 16; ++i) {
+    std::cout << input0_data[i] << " + " << input1_data[i] << " = "
+              << output0_data[i] << std::endl;
+    std::cout << input0_data[i] << " - " << input1_data[i] << " = "
+              << output1_data[i] << std::endl;
+
+    if ((input0_data[i] + input1_data[i]) != output0_data[i]) {
+      std::cerr << "error: incorrect sum" << std::endl;
+      exit(1);
+    }
+    if ((input0_data[i] - input1_data[i]) != output1_data[i]) {
+      std::cerr << "error: incorrect difference" << std::endl;
+      exit(1);
+    }
+  }
+
+  // Get shared memory regions all active/registered within TRTIS
+  ni::SharedMemoryStatus status;
+  err = shared_memory_ctx->GetSharedMemoryStatus(&status);
+  if (!err.IsOk()) {
+    std::cerr << "error: " << err << std::endl;
+    exit(1);
+  }
+  std::cout << "Shared Memory Status:\n" << status.DebugString() << "\n";
+
+  // Unregister shared memory (One by one or all at a time) from TRTIS
+  // err = shared_memory_ctx->UnregisterAllSharedMemory();
+  err = shared_memory_ctx->UnregisterSharedMemory("input_data");
+  if (!err.IsOk()) {
+    std::cerr << "error: unable to unregister shared memory input region: "
+              << err << std::endl;
+    exit(1);
+  }
+  err = shared_memory_ctx->UnregisterSharedMemory("output_data");
+  if (!err.IsOk()) {
+    std::cerr << "error: unable to unregister shared memory output region: "
+              << err << std::endl;
+    exit(1);
+  }
+
+  // cleanup cuda shared memory
+  checkCudaErrors(cudaIpcCloseMemHandle(input0_d_ptr));
+  checkCudaErrors(cudaIpcCloseMemHandle(input1_d_ptr));
+
+  // Cleanup system shared memory
+  UnmapSharedMemory(input0_cuda_rep, cuda_ipc_byte_size * 2);
+  UnlinkSharedMemoryRegion("/input_simple");
+  UnmapSharedMemory(output0_cuda_rep, cuda_ipc_byte_size * 2);
+  UnlinkSharedMemoryRegion("/output_simple");
```

Comment:
missing `output1_cuda_rep`

Reply:
output1_cuda_rep is deleted at the same time since output1_cuda_rep and output0_cuda_rep both reside in continuous regions in the same shared memory region. Similarly for input0_cuda_rep and input1_cuda_rep.